vk_generator/src/generator/mod.rs

```diff
@@ -940,17 +995,21 @@ impl<'a> GenTypes<'a> {
                     use self::ConstType::*;
 
                     let consts = &mut gen_types.consts;
-                    let (name, value) = unsafe{ (&*name, (&*value).trim()) };
+                    let (mut name, value) = unsafe{ (&*name, (&*value).trim()) };
                     let mut typ = processed.const_types.get(name).map(|t| *t).unwrap_or(Unknown);
                     let mut slice_indices = (0, value.len());
 
-                    // Ignore enum variants that have been renamed in the Vulkan specs.
-                    if typ == Unknown &&
-                       name.starts_with("VK_") &&
-                       value.starts_with("VK_") {
-                        continue;
+                    if name.starts_with("VK_") {
+                        // Ignore enum variants that have been renamed in the Vulkan specs.
+                        if typ == Unknown && value.starts_with("VK_") {
+                            continue;
+                        }
+                        if gen_types.config.remove_type_prefix {
+                            name = &name[3..];
+                        }
```

Comment:
Move the `const` variant stripping to `add_interface`, in the `ConstDef` branch. Again, I don't think `const` identifier stripping should be handled by `remove_type_prefix`. It's distinct enough from type stripping and enum stripping that I think it deserves its own flag.